src/typed_ast.cpp

```diff
@@ -59,24 +60,21 @@ TypedAST* convert_ast(AST::DictionaryLiteral* ast, Allocator& alloc) {
 TypedAST* convert_ast(AST::FunctionLiteral* ast, Allocator& alloc) {
 	auto typed_function = alloc.make<FunctionLiteral>();
 
-	for (auto arg : ast->m_args) {
-		assert(arg->type() == ASTTag::Declaration);
-		auto decl = static_cast<AST::Declaration*>(arg);
-
-		auto typed_decl = alloc.make<Declaration>();
+	for (auto& arg : ast->m_args) {
+		Declaration typed_decl;
 
-		typed_decl->m_identifier_token = decl->m_identifier_token;
-		typed_decl->m_surrounding_function = typed_function;
+		typed_decl.m_identifier_token = arg.m_identifier_token;
+		typed_decl.m_surrounding_function = typed_function;
 
-		typed_function->m_args.push_back(typed_decl);
+		typed_function->m_args.push_back(std::move(typed_decl));
 	}
 
 	typed_function->m_body = convert_ast(ast->m_body, alloc);
 
 	return typed_function;
 }
 
-Declaration* convert_ast(AST::Declaration* ast, Allocator& alloc) {
+TypedAST* convert_ast(AST::Declaration* ast, Allocator& alloc) {
```

Comment:
That's weird.. why does this not return a `Declaration*`?

Reply:
I reckon there was a type error. I supposed that it was because I was changing the return type. I find it weird too because I was sure it was like that before.